lib/eex/lib/eex/compiler.ex

```diff
@@ -58,7 +57,22 @@ defmodule EEx.Compiler do
     {marker, t} = retrieve_marker(t)
 
     case expr(t, line, column + 2 + length(marker), state, []) do
-      {:error, line, column, message} ->
+      {:error, _line, _column, message} ->
+        code_snippet = code_snippet(state.source, %{line: line, column: column}, 0)
```

Comment:
```suggestion
        code_snippet = code_snippet(state.source, marker, %{line: line, column: column}, 0)
```

And then use `length(marker)` in `code_snippet`. :)

Reply:
We send the arrow padding as last arg. So just sending the marker_length seems there seems to do the work.